src/plugins/faultManagement/FaultManagementListView.vue

```diff
@@ -0,0 +1,333 @@
+        },
+        async toggleShelveSelected(faults = Object.values(this.selectedFaults), shelveData = {}) {
+            const { shelved = true } = shelveData;
+            if (shelved) {
+                let title = faults.length > 1
+                    ? `Shelve ${faults.length} selected faults`
+                    : `Shelve fault: ${faults[0].id.name}`
+                ;
+
+                const formStructure = {
+                    title,
+                    sections: [
+                        {
+                            rows: [
+                                {
+                                    key: 'comment',
+                                    control: 'textarea',
+                                    name: 'Comment',
+                                    pattern: '\\S+',
+                                    required: false,
+                                    cssClass: 'l-input-lg',
+                                    value: ''
+                                },
+                                {
+                                    key: 'shelveDuration',
+                                    control: 'select',
+                                    name: 'Shelve Duration',
+                                    options: FAULT_MANAGEMENT_SHELVE_DURATIONS_IN_MS,
+                                    required: false,
+                                    cssClass: 'l-input-lg',
+                                    value: FAULT_MANAGEMENT_SHELVE_DURATIONS_IN_MS[0].value
+                                }
+                            ]
+                        }
+                    ],
+                    buttons: {
+                        submit: {
+                            label: 'Shelve'
+                        }
+                    }
+                };
+
+                let data;
+                try {
+                    data = await this.openmct.forms.showForm(formStructure);
+                } catch (e) {
+                    // Do nothing
+                    return;
+                }
+
+                shelveData.comment = data.comment || '';
+                shelveData.shelveDuration = data.shelveDuration !== undefined
+                    ? data.shelveDuration
+                    : FAULT_MANAGEMENT_SHELVE_DURATIONS_IN_MS[0].value;
+            } else {
+                shelveData = {
+                    shelved: false
+                };
+            }
+
+            Object.values(faults)
+                .forEach(selectedFault => {
+                    this.openmct.faults.shelveFault(selectedFault, shelveData);
+                });
+
+            this.selectedFaults = {};
+        },
+        updateFilter(filter) {
+            this.selectAll();
+
+            this.filterIndex = filter.model.options.findIndex(option => option.value === filter.value);
+        },
+        updateSearchTerm(term = '') {
+            this.searchTerm =  term.toLowerCase();
+        }
+    }
+};
+</script>
+
+<style>
```

Comment:
Are these supposed to be here? Our current build tooling (unfortunately) does not support styles in SFCs, they need to go into a separate `scss` file that is reachable from the webpack entry point.